### PR TITLE
Changes During PNW Champs

### PR DESCRIPTION
**Pull request by RobinsonZ**
that's what this is

### CODE REVIEW

File: profiles/trajectory.properties
Comment (edelmanjm) [inline profiles/trajectory.properties:7]: Is this just an experimental value? If so, where'd it come from?

File: src/main/java/org/team1540/robot2018/OI.java
Comment (edelmanjm) [inline src/main/java/org/team1540/robot2018/OI.java:165]: Deal with this then

File: src/main/java/org/team1540/robot2018/OI.java
Comment (edelmanjm) [inline src/main/java/org/team1540/robot2018/OI.java:181]: Why commented blocks of code?

File: src/main/java/org/team1540/robot2018/Robot.java
Comment (edelmanjm) [inline src/main/java/org/team1540/robot2018/Robot.java]: Use enums rather than Strings

File: src/main/java/org/team1540/robot2018/Robot.java
Comment (edelmanjm) [inline src/main/java/org/team1540/robot2018/Robot.java:101]: Can we centralize all the SmartDashboard putting? It's awfully annoying to have it spread around.

File: src/main/java/org/team1540/robot2018/Robot.java
Comment (edelmanjm) [inline src/main/java/org/team1540/robot2018/Robot.java]: Surely this could be made way, way, way nicer using enums and methods around those?

File: src/main/java/org/team1540/robot2018/Robot.java
Comment (edelmanjm) [inline src/main/java/org/team1540/robot2018/Robot.java:196]: Should decide on a consistent naming scheme for these, as given the lack of organization in SmartDashboard/Shuffleboard, it can be terribly hard to find things.

File: src/main/java/org/team1540/robot2018/commands/TankDrive.java
Comment (RobinsonZ) [inline src/main/java/org/team1540/robot2018/commands/TankDrive.java:10]: This should be in `initialize()`, not the constructor.

File: src/main/java/org/team1540/robot2018/commands/JoystickTape.java
Comment (edelmanjm) [inline src/main/java/org/team1540/robot2018/commands/JoystickTape.java:17]: no what why

File: src/main/java/org/team1540/robot2018/commands/arms/OpenArms.java
Comment (edelmanjm) [inline src/main/java/org/team1540/robot2018/commands/arms/OpenArms.java:7]: *insert pun here*

File: src/main/java/org/team1540/robot2018/commands/auto/DriveTimed.java
Comment (edelmanjm) [inline src/main/java/org/team1540/robot2018/commands/auto/DriveTimed.java:29]: ...is that very hard?

File: src/main/java/org/team1540/robot2018/commands/arms/OpenArms.java
Comment (RobinsonZ) [inline src/main/java/org/team1540/robot2018/commands/arms/OpenArms.java:14]: Conflicts with the `DropCube` command; do we invert one side to open, or not?

File: src/main/java/org/team1540/robot2018/commands/auto/sequences/SwitchDoubleCube.java
Comment (edelmanjm) [inline src/main/java/org/team1540/robot2018/commands/auto/sequences/SwitchDoubleCube.java:18]: This works, but I feel like there's gotta be a better way of doing this hierarchy from an end user ish standpoint.

File: src/main/java/org/team1540/robot2018/commands/auto/sequences/SwitchDoubleCube.java
Comment (RobinsonZ) [inline src/main/java/org/team1540/robot2018/commands/auto/sequences/SwitchDoubleCube.java:36]: Should not be sequential. However, removing this might cause wrist to hit the switch fence–should be tested

File: src/main/java/org/team1540/robot2018/motion/FollowProfile.java
Comment (edelmanjm) [inline src/main/java/org/team1540/robot2018/motion/FollowProfile.java:1]: Honestly can't properly review this without a lot more time and effort, so a lot of trust here. Also put this in ROOSTER at some point SMH.

File: src/main/java/org/team1540/robot2018/commands/auto/sequences/SwitchDoubleCube.java
Comment (RobinsonZ) [inline src/main/java/org/team1540/robot2018/commands/auto/sequences/SwitchDoubleCube.java:52]: Does nothing.

File: src/main/java/org/team1540/robot2018/motion/FollowProfile.java
Comment (edelmanjm) [inline src/main/java/org/team1540/robot2018/motion/FollowProfile.java:3]: eh i guess

File: src/main/java/org/team1540/robot2018/motion/FollowProfile.java
Comment (edelmanjm) [inline src/main/java/org/team1540/robot2018/motion/FollowProfile.java:69]: reeee i want my blame /s

File: src/main/java/org/team1540/robot2018/motion/FollowProfile.java
Comment (edelmanjm) [inline src/main/java/org/team1540/robot2018/motion/FollowProfile.java:99]: uh

File: src/main/java/org/team1540/robot2018/commands/auto/sequences/SwitchDoubleCube.java
Comment (RobinsonZ) [inline src/main/java/org/team1540/robot2018/commands/auto/sequences/SwitchDoubleCube.java:27]: This can be added in parallel, though it doesn't really matter (has the same effect).

**Review by edelmanjm — DISMISSED** [review]
Honestly mostly just enums for auto

File: src/main/java/org/team1540/robot2018/motion/FollowProfile.java
Comment (RobinsonZ) [reply src/main/java/org/team1540/robot2018/motion/FollowProfile.java:99]: IntelliJ inspection, it was confused that I was passing the setpoint Y to a parameter called X

Comment (edelmanjm) [reply src/main/java/org/team1540/robot2018/motion/FollowProfile.java:99]: Does it not use annotations?

File: src/main/java/org/team1540/robot2018/motion/FollowProfile.java
Comment (RobinsonZ) [reply src/main/java/org/team1540/robot2018/motion/FollowProfile.java:1]: You should be able to trust it considering we ran it all of PNWs lol

And yes I will but that's a lot of effort

File: src/main/java/org/team1540/robot2018/commands/auto/sequences/SwitchDoubleCube.java
Comment (RobinsonZ) [reply src/main/java/org/team1540/robot2018/commands/auto/sequences/SwitchDoubleCube.java:18]: we are lowkey the end user tho

File: src/main/java/org/team1540/robot2018/motion/FollowProfile.java
Comment (edelmanjm) [reply src/main/java/org/team1540/robot2018/motion/FollowProfile.java:1]: Working code != good code

File: src/main/java/org/team1540/robot2018/Robot.java
Comment (RobinsonZ) [reply src/main/java/org/team1540/robot2018/Robot.java]: yes if you want to do that be my guest

File: src/main/java/org/team1540/robot2018/commands/auto/sequences/SwitchDoubleCube.java
Comment (edelmanjm) [reply src/main/java/org/team1540/robot2018/commands/auto/sequences/SwitchDoubleCube.java:18]: Point is that we want things to be easier to work with.

File: src/main/java/org/team1540/robot2018/Robot.java
Comment (edelmanjm) [reply src/main/java/org/team1540/robot2018/Robot.java]: Will do later this week, I have lots of HW.

File: src/main/java/org/team1540/robot2018/Robot.java
Comment (RobinsonZ) [reply src/main/java/org/team1540/robot2018/Robot.java:101]: It's all in `robotInit()` for stuff that's set-and-forget and `robotPeriodic()` for telemetry, don't know how more centralized you want to be

Comment (edelmanjm) [reply src/main/java/org/team1540/robot2018/Robot.java:101]: Putting it in the same parts of said methods.

File: profiles/trajectory.properties
Comment (RobinsonZ) [reply profiles/trajectory.properties:7]: It's closer to our "effective" width considering wheel scrub. We ran some testing involving turning the robot in a circle a lot and measuring the distance.

Comment (edelmanjm) [reply profiles/trajectory.properties:7]: oh hey it's that dude's suggestion

**Review by liamjwang — APPROVED** [review]

**Comment by edelmanjm** [comment]
@wangl5 There's a bunch of things that can and should be easily addressed though. Seriously even just run through @RobinsonZ's stuff.

**Review by edelmanjm — DISMISSED** [review]
Don't dismiss me and Zack

**Comment by wangl5** [comment]
@edelmanjm Zack said to

**Comment by edelmanjm** [comment]
@wangl5 Just make those changes before merge, or if not, I can

**Comment by liamjwang** [comment]
@edelmanjm Zack said the changes already happened tho and I should merge it

**Comment by RobinsonZ** [comment]
@edelmanjm Okay so long story short we're gonna merge this (as it's the latest working copy) and create a dev branch for Houston immediately after. Given that most of your changes are stuff that has been fixed by your PR and most of mine are stuff that I officially don't care that much about we're merging this.

**Comment by edelmanjm** [comment]
¯\\\_(ツ)_/¯ fine

**Pull request merged**
